src/util/charmrun-src/charmrun.C

```diff
@@ -5399,6 +5417,41 @@ static void start_nodes_mpiexec()
   /* all ssh_pid remain zero: skip finish_nodes */
 }
 
+static int finish_one_node(nodetab_process & p, int & retries, const char ** argv = arg_argv)
```

Comment:
What's the meaning of the return value?

Reply:
It's an idiom where each non-success return site uses a different non-zero value. The return value is only used in a loop checking that it non-zero.